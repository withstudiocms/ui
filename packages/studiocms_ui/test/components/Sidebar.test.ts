import { describe, expect } from 'vitest';
import DoubleSidebar from '../../src/components/Sidebar/Double.astro';
import SingleSidebar from '../../src/components/Sidebar/Single.astro';
import { test } from '../fixtures/vitest/AstroContainer';

const mockContent = '<p>This is some sidebar content.</p>';

describe('Sidebar Component', () => {
	test('renders SingleSidebar component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(SingleSidebar, 'SingleSidebar', {
			slots: { default: mockContent },
		});
		expect(result).toMatchSnapshot();
	});

	test('renders DoubleSidebar component correctly', async ({ renderComponent }) => {
		const result = await renderComponent(DoubleSidebar, 'DoubleSidebar', {
			slots: { inner: mockContent, outer: mockContent },
		});
		expect(result).toMatchSnapshot();
	});
});
