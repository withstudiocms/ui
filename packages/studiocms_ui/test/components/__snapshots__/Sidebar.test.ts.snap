// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar Component > renders DoubleSidebar component correctly 1`] = `"<div id="sui-sidebars" class="active inner"> <div id="sui-sidebar-outer"> <p>This is some sidebar content.</p> </div> <div id="sui-sidebar-inner"> <p>This is some sidebar content.</p> </div> </div> "`;

exports[`Sidebar Component > renders SingleSidebar component correctly 1`] = `"<aside id="sui-sidebar" class="astro-visbj6pb"> <p>This is some sidebar content.</p> </aside> "`;
